File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.js
/**
 * @param {string[]} words
 * @return {string}
 */
var firstPalindrome = function(words) {
  const isPal = s => s.length < 2 || (s[0] === s.at(-1) && isPal(s.slice(1, -1)))
  for (let word of words) {
    if (isPal(word)) return word
  }
  return ""
};

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.go
func isPalindrome(s string) bool {
  l, r := 0, len(s) - 1
  for l < r {
    if s[l] != s[r] {
      return false
    }
    l++
    r--
  }
  return true
}

func firstPalindrome(words []string) string {
  for _, word := range words {
    if isPalindrome(word) {
      return word
    }
  }  
  return ""
}